### ISSUE TITLE
Export lightweight ABIs

### ISSUE THREAD

**Issue by lucaperret** [ItsNickBarry/hardhat-abi-exporter#20]
Hi there, thanks for this plugin!

Would be great to have the option to export only some functions and events to have lighter ABIs...
Is this feature would makes sense on this plugin? What do you recommend to implement it?
It'd be happy to contribute on this one.

Best regards,
Luca

**Comment by ItsNickBarry** [comment]
Have you tried the `pretty` option?  It exports interface style ABIs, which have a smaller file size and are compatible with ethers.js (not sure if other libraries support them).

If you're asking about removing specific unused items from the ABI before export, I think that would be a mistake for two reasons:
1. The configuration required for this would be very verbose and inconvenient.
2. If one removes functions from a standard ABI such as `ERC20.json`, the file no longer matches the standard.

**Comment by lucaperret** [comment]
1. I totally agree, hence my request for suggestion here. We could potentially export only the "external" and "public" functions + events for instance...
2. Considering that generally the front-end only need to interact with some of the functions, it could reduce a lot the exported ABI for the front-end target.

**Comment by ItsNickBarry** [comment]
ABIs only contain public and external items, unless I'm mistaken.  If you see a private function in an ABI, let me know.

If you're worried about ABI file size, I would recommend an approach like this:
https://github.com/tsla-exchange/tsla-exchange-ui/blob/3b892b77cfc0e68c47fe697111efda7712e47336/pages/modal.vue#L193-L200

But if you really want to use this plugin to export partial ABIs, you can override the compile task to modify the files to suit your needs:
```javascript
const {
  TASK_COMPILE,
} = require('hardhat/builtin-tasks/task-names');

task(TASK_COMPILE, async function (args, hre, runSuper) {
  await runSuper();
  // TODO: read ABI json, delete unwanted properties, overwrite ABI json
});
```

**Comment by lucaperret** [comment]
Indeed, I was referring to the functions for the end users.

Thanks for your insights @ItsNickBarry, I think I'll create a separate plugin for this need.

**Comment by ItsNickBarry** [comment]
@lucaperret I think we might be adding a generic `filter` function to the configuration that will allow ABI items to be removed from the final export.  Because it seems like there are multiple use cases, it might make sense to add support.

**Comment by lucaperret** [comment]
That'd be perfect!
Thanks @ItsNickBarry

**Comment by ItsNickBarry** [comment]
Published in version `2.6.0`.
